Remotion/Data/DomainObjects/Linq/DomainObjectQueryGenerator.cs

```diff
@@ -274,9 +274,10 @@ private IRelationEndPointDefinition GetEagerFetchRelationEndPointDefinition (Fet
       }
 
       var propertyInfoAdapter = PropertyInfoAdapter.Create(propertyInfo);
-      var endPoint = typeDefinition.GetTypeHierarchy()
-                         .Select(cd => cd.ResolveRelationEndPoint(propertyInfoAdapter))
-                         .FirstOrDefault(ep => ep != null);
+      var endPoint = InlineTypeDefinitionWalker.WalkDescendants(
```

Comment:
Here's a good example where the api no longer communicates that the walk stops once the endpoint is found.